src/hotspot/share/oops/generateOopMap.cpp

```diff
@@ -906,12 +907,15 @@ void GenerateOopMap::monitor_push(CellTypeState cts) {
 // Interpretation handling methods
 //
 
-void GenerateOopMap::do_interpretation()
+void GenerateOopMap::do_interpretation(Thread* thread)
 {
-  // "i" is just for debugging, so we can detect cases where this loop is
-  // iterated more than once.
   int i = 0;
   do {
+    if (i != 0 && thread->is_Java_thread()) {
+      if (thread->as_Java_thread()->thread_state() == _thread_in_vm) {
+        ThreadBlockInVM tbivm(thread->as_Java_thread());
+      }
+    }
```

Comment:
Can you add a comment as to why this is necessary please.

Reply:
Perhaps something like this above L916:

```
  // Since this JavaThread has looped at least once and is _thread_in_vm,
  // we honor any pending blocking request.

Reply:
Fixed